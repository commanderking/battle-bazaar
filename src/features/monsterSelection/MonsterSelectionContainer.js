import React from 'react';
import { monstersData } from '../../data/monsters';
import { MonsterSelect } from './MonsterSelect';
import { MonsterCard } from './MonsterCard';
import { observer } from 'mobx-react';
import { battlePhases } from '../../constants/battleConstants';

@observer
class MonsterSelectionContainer extends React.Component {
  render() {
    const { appStore } = this.props;
    return (
      <div>
        <MonsterSelect monstersData={monstersData} appStore={appStore} />
        { appStore.selectedMonsterName &&
          <div>
            <MonsterCard appStore={appStore} />
            <button onClick={() => {
                appStore.changeBattlePhase(battlePhases.CHARACTER_SELECT)
            }}
            >
              Fight Monster
            </button>
          </div>
        }
      </div>
    )
  }
}

export { MonsterSelectionContainer };
