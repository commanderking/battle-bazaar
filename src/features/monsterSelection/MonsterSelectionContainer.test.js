import React from 'react';
import ReactDOM from 'react-dom';
import { MonsterSelectionContainer } from './MonsterSelectionContainer';
import { shallow } from 'enzyme';

describe('MonsterSelectionContainer', () => {
  it('renders and matches snapshot', () => {
    const wrapper = shallow(<MonsterSelectionContainer />);
    expect(wrapper).toMatchSnapshot();
  });
});
