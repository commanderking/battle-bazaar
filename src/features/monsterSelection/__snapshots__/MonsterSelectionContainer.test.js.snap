// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonsterSelectionContainer renders and matches snapshot 1`] = `
<div>
  Placeholder for Monster Selection
</div>
`;
